### ISSUE TITLE
"format" request to tsserver does not seem to respect options

### ISSUE THREAD

**Issue by gins3000** [theia-ide/typescript-language-server#41]
Observed in version 1.11
Observed with tsserver both from typescript version 2.7.2 (specified in package.json) and current 2.8.x

In our project, we want the user to somehow configure their formatting preferences (e.g. tabs or spaces and their size), but the format request to tsserver seems to ignore the options that get passed in, regardless of if tsfmt is used, or the options coming in from the language server protocol (which we would prefer).

As an example, when putting a tsfmt.json file in the project root as described in [1], I have tracked that this file is properly read and the parsed options passed to tsserver at this point[2], but tsserver seems to ignore all options, in particular the tab and indent sizes. Instead it always goes with 4 spaces.

Is there an issue in the way you pass the options to tsserver or is tsserver at fault?

tsfmt.json file I used:
```
{
  "tabSize": 2,
  "indentSize": 2,
  "convertTabsToSpaces": true
}
```
[1] https://github.com/theia-ide/typescript-language-server/issues/10
[2] https://github.com/theia-ide/typescript-language-server/blob/92ccb07fec4fc8a4c76c89b061ab8a31f808c7a4/src/lsp-server.ts#L428

**Comment by svenefftinge** [comment]
options are not support atm, but tsfmt.json should work. If not it is a bug.

**Comment by gins3000** [comment]
Thanks for the fast response.

As said in the issue description, even tsfmt.json does not work as expected. So that sounds like a bug, or I'm missing something.

When using the tsfmt.json above, which says to use 2 spaces, the format request still applies 4 spaces, not 2. If I put in random numbers, I get the same result, so the tsfmt.json does not seem to have an influence on the outcome.

**Comment by gins3000** [comment]
A small update:

This line in the code of tsserver seems suspicious ([1]).
It seems like typescript-language-server implements the protocol correctly, but tsserver doesn't support it :(

There does seem to be an alternative way to provide formatting options with a "configure" command[2], which I'm looking into now.

[1] https://github.com/Microsoft/TypeScript/blob/f35a20e29e3f74e5c049855db05471430368ce49/src/server/client.ts#L240
[2] https://github.com/Microsoft/TypeScript/blob/ea2bc4c0141992e0172cf2fea80693807458eaac/src/server/protocol.ts#L1308

**Comment by gins3000** [comment]
Thanks to the PR being accepted, this can be considered closed :)

**Comment by svenefftinge** [comment]
I published 0.1.12 with the changes. Please verify.

**Comment by gins3000** [comment]
Seems to work, thanks! :+1:
